File: 23/kod.py
def d(number):

    i, result = 1, 0

    while i*i < number:
        if number % i == 0:
            result += i + number//i
        i += 1

    if i*i == number:
        result += i

    return result-number


limit, result = 28123, 0
abundant, sums = [], [False]*(limit+1)

for i in range(12, limit):
    if d(i) > i:
        abundant.append(i)

for i in range(len(abundant)):
    for j in range(i, len(abundant)):
        if abundant[i] + abundant[j] < limit:
            sums[abundant[i]+abundant[j]] = True

for i in range(limit):
    if sums[i] == False:
        result += i

print(result)

File: 124/kod.cpp
#include <cstdio>
#include <iostream>
#include <vector>

using namespace std;

int N, K;
vector<pair<int, int>> rads;

inline int rad( int number ){

    int limit = number, result = 1;

    for( int i=2 ; i*i <= limit ; i++ )
        if( number % i == 0 ){

            result *= i;

            while (number % i == 0 )
                number /= i;
        }
    
    return result*number;
}

int main(){

    scanf("%d%d", &N, &K);
    
    for( int i=1 ; i<=N ; i++ )
        rads.push_back(pair<int, int>(rad(i), i));

    sort(rads.begin(), rads.end());

    printf("%d\n", rads[K-1].second);
    
    return 0;
}
